### PR TITLE
Use the lowest version from a running resource, not just the lowest

### PR DESCRIPTION
**Pull request by tobio**
## Description

Updates the deployment version determination to filter out stopped resources. The API returns the current plan for resources which have at some point been running, which should not be considered when finding the lowest version.  

## Related Issues

Fixes https://github.com/elastic/terraform-provider-ec/issues/614

## How Has This Been Tested?

Updated an existing acceptance test to cover this scenario

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/acc/deployment_post_node_role_upgrade_test.go
Comment (dimuon) [inline ec/acc/deployment_post_node_role_upgrade_test.go:1]: @tobio , I suppose this test should fail without the fix (without the fix it should output `Error: Provider produced inconsistent result after apply` complaining about the version), correct?

**Review by dimuon — APPROVED** [review]
LGTM

**Comment by dimuon** [comment]
@tobio , I think both unit and integration tests fail due to changes in core terraform, so if we use an older terraform cli (something like 1.3.9). the tests should pass. Anyway we need to address the issue ASAP but I believe we have also to move to the latest Framework version to handle that correctly.

**Comment by tobio** [comment]
Yep, I fixed the TF version in #139

File: ec/acc/deployment_post_node_role_upgrade_test.go
Comment (tobio) [reply ec/acc/deployment_post_node_role_upgrade_test.go:1]: Yep

**Pull request merged**